manager/controllers/app/np_isolation_read_flow_test.go

```diff
@@ -0,0 +1,376 @@
+		ToNot(gomega.HaveOccurred())
+	application.ObjectMeta.Name += "-1"
+	application.Spec.Data[0].DataSetID = catalogedAsset
+	applicationKey = client.ObjectKeyFromObject(application)
+
+	// Create FybrikApplication
+	fmt.Println("Expecting application creation to succeed")
+	g.Expect(k8sClient.Create(context.Background(), application)).Should(gomega.Succeed())
+
+	// Ensure getting cleaned up after tests finish
+	// delete application
+	defer func() {
+		fybrikApplication := &fapp.FybrikApplication{ObjectMeta: metav1.ObjectMeta{Namespace: applicationKey.Namespace,
+			Name: applicationKey.Name}}
+		_ = k8sClient.Get(context.Background(), applicationKey, fybrikApplication)
+		_ = k8sClient.Delete(context.Background(), fybrikApplication)
+	}()
+
+	fmt.Println("Expecting application to be created")
+	g.Eventually(func() error {
+		return k8sClient.Get(context.Background(), applicationKey, application)
+	}, timeout, interval).Should(gomega.Succeed())
+	fmt.Println("Expecting plotter to be constructed")
+	g.Eventually(func() *fapp.ResourceReference {
+		_ = k8sClient.Get(context.Background(), applicationKey, application)
+		return application.Status.Generated
+	}, timeout, interval).ShouldNot(gomega.BeNil())
+
+	// The plotter has to be created
+	plotterObjectKey = client.ObjectKey{Namespace: application.Status.Generated.Namespace,
+		Name: application.Status.Generated.Name}
+	fmt.Println("Expecting plotter to be fetchable")
+	g.Eventually(func() error {
+		return k8sClient.Get(context.Background(), plotterObjectKey, plotter)
+	}, timeout, interval).Should(gomega.Succeed())
+
+	fmt.Println("Expecting application to be ready")
+	g.Eventually(func() bool {
+		err = k8sClient.Get(context.Background(), applicationKey, application)
+		if err != nil {
+			return false
+		}
+		return application.Status.Ready
+	}, timeout, interval).Should(gomega.Equal(true))
+
+	modulesNamespace = plotter.Spec.ModulesNamespace
+
+	g.Expect(application.Status.AssetStates[catalogedAsset].Endpoint.Name).ToNot(gomega.BeEmpty())
+	g.Expect(application.Status.AssetStates[catalogedAsset].Conditions[ReadyConditionIndex].Status).To(gomega.Equal(v1.ConditionTrue))
+
+	// Get the connection endpoint
+	connection := application.Status.AssetStates[catalogedAsset].
+		Endpoint.AdditionalProperties.Items["fybrik-arrow-flight"].(map[string]interface{})
+	hostname := fmt.Sprintf("%v", connection["hostname"])
+	port := fmt.Sprintf("%v", connection["port"])
+
+	// using my-shell pod to read
+	gvk := schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Pod",
+	}
+
+	restClient, err := apiutil.RESTClientForGVK(gvk, false, ctrl.GetConfigOrDie(), serializer.NewCodecFactory(scheme.Scheme))
+	g.Expect(err).To(gomega.BeNil())
+
+	readCommand := "python3 /root/client.py --host " + hostname + " --port " + port + " --asset " + catalogedAsset
+	var stdout, stderr bytes.Buffer
+	fmt.Println("Expecting successfull read from the namesapce of the module")
```

Comment:
"Expecting successfully read from the namespace of the module"

Reply:
I doubt that it will work, the fybrik-blueprints namespace has a network policy `fybrik-blueprints-default-deny-all` that prevents all unapproved connections to and from all pods.

Reply:
Fixed.